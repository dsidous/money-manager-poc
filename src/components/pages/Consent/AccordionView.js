import React, { PureComponent } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import Accordion from 'react-native-collapsible/Accordion';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#F5FCFF',
  },
  header: {
    backgroundColor: '#d6d6d6',
    marginBottom: 5,
    padding: 10,
  },
  headerText: {
    textAlign: 'left',
    fontSize: 18,
    fontWeight: '600',
  },
  content: {
    padding: 20,
    backgroundColor: '#f1f1f1',
  },
  contentText: {
    fontSize: 18,
  },
  active: {
    backgroundColor: 'rgba(255,255,255,1)',
  },
  inactive: {
    backgroundColor: 'rgba(245,252,255,1)',
  },
  selectors: {
    marginBottom: 10,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  selector: {
    backgroundColor: '#F5FCFF',
    padding: 10,
  },
  activeSelector: {
    fontWeight: 'bold',
  },
  selectTitle: {
    fontSize: 16,
    fontWeight: '500',
    padding: 10,
  },
});

const SECTIONS = [
  {
    title: 'Your account details',
    content: `We will have read access to :
      -  Your bank account details like account name, account number and sort code
      -  Your account balance and overdraft limits, available balance`,
  },
  {
    title: 'Your last 12 months account details',
    content: `-  Details of your incoming transactions 
    
-  Details of your outgoing transactions
    `,
  },
];

class AccordionView extends PureComponent {
  renderHeader(section) {
    return (
      <View style={styles.header}>
        <Text style={styles.headerText}>{section.title}</Text>
      </View>
    );
  }

  renderContent(section) {
    return (
      <View style={styles.content}>
        <Text style={styles.contentText}>{section.content}</Text>
      </View>
    );
  }

  render() {
    return (
      <Accordion
        sections={SECTIONS}
        renderHeader={this.renderHeader}
        renderContent={this.renderContent}
      />
    );
  }
}

export default AccordionView;