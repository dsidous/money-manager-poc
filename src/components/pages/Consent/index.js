import React, { PureComponent } from 'react';
import { StyleSheet, Switch, ScrollView, View } from 'react-native';
import { StyledButton, StyledView, StyledText } from '../../atoms';
import AccordionView from './AccordionView';
import PropTypes from 'prop-types';

const StyledContainer = StyledView.extend`
  flex-direction: row;
  align-items: center;
  justify-content: center;
`

export default class Consent extends PureComponent {

  state = {
    nextButtonEnabled: false
  }

  switchValueChange = () => this.setState((prevState) => ({ nextButtonEnabled: !prevState.nextButtonEnabled }));

  render() {
    const { navigation } = this.props;

    return (
      <ScrollView style={{ paddingHorizontal: 25, marginTop: 20, }}>
        <StyledText style={{ marginBottom: 15 }} size='large'>
          Your consent
        </StyledText>
        <View style={{ marginBottom: 15, flex: 1 }}>
          <StyledText weight>Don't worry, this is all perfectly safe.</StyledText>
          <StyledText weight>All we can do is look at...</StyledText>
        </View>
        <View>
          <AccordionView />
        </View>
        <View>
          <StyledText style={{ marginVertical: 15 }}>
            gameplan can only read your bank data and can't make payments or change your account.
          </StyledText>
        </View>
        <View>
          <StyledText style={{ marginBottom: 15 }}>
            We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission
          </StyledText>
        </View>
        <StyledContainer>
          <StyledView style={{ alignItems: 'center' }}>
            <Switch
              onValueChange={this.switchValueChange}
              value={this.state.nextButtonEnabled}
            />
          </StyledView>
          <StyledText style={{ fontSize: 15, flex: 2, marginBottom: 15 }}>
            You agree to our terms &amp; conditions and privacy policy which covers how giffgaff can access and use your data
          </StyledText>
        </StyledContainer>
        <View>
          <StyledText style={{ marginBottom: 15 }}>
            We will now securely take you to your bank/building society
          </StyledText>
        </View>
        <StyledView style={{ flexDirection: 'row', padding: 30 }}>
          <StyledButton
            outlined
            style={{ flex: 1, marginRight: 30 }}
            onPress={() => navigation.navigate('ChooseBank')}
          >
            Disagree
          </StyledButton>
          <StyledButton
            disabled={!this.state.nextButtonEnabled}
            style={{ flex: 1 }}
            onPress={() => navigation.navigate('ChooseAccount')}
          >
            Agree
          </StyledButton>
        </StyledView>
      </ScrollView>
    );
  }
}

Consent.propTypes = {
  navigation: PropTypes.object,
}