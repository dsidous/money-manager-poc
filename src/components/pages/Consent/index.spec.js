import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Consent from './';

configure({ adapter: new Adapter() });

describe('consent', () => {
  const navigation = { navigate: jest.fn() };
  const render = shallow(<Consent navigation={navigation} />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('Test switch change event', () => {
    render.find('Switch').simulate('valueChange');
    render.find('StyledButton').at(1).simulate('press');
    expect(navigation.navigate).toHaveBeenCalledWith('ChooseAccount');
  });
});
