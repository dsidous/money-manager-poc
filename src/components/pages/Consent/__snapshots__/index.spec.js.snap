// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`consent renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Consent
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StyledText
          size="large"
          style={
            Object {
              "marginBottom": 15,
            }
          }
        >
          Your consent
        </StyledText>,
        <Component
          style={
            Object {
              "flex": 1,
              "marginBottom": 15,
            }
          }
        >
          <StyledText
            weight={true}
          >
            Don't worry, this is all perfectly safe.
          </StyledText>
          <StyledText
            weight={true}
          >
            All we can do is look at...
          </StyledText>
        </Component>,
        <Component>
          <AccordionView />
        </Component>,
        <Component>
          <StyledText
            style={
              Object {
                "marginVertical": 15,
              }
            }
          >
            gameplan can only read your bank data and can't make payments or change your account.
          </StyledText>
        </Component>,
        <Component>
          <StyledText
            style={
              Object {
                "marginBottom": 15,
              }
            }
          >
            We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission
          </StyledText>
        </Component>,
        <Styled(Component)>
          <Styled(Component)
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Switch
              disabled={false}
              onValueChange={[Function]}
              value={false}
            />
          </Styled(Component)>
          <StyledText
            style={
              Object {
                "flex": 2,
                "fontSize": 15,
                "marginBottom": 15,
              }
            }
          >
            You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data
          </StyledText>
        </Styled(Component)>,
        <Component>
          <StyledText
            style={
              Object {
                "marginBottom": 15,
              }
            }
          >
            We will now securely take you to your bank/building society
          </StyledText>
        </Component>,
        <Styled(Component)
          style={
            Object {
              "flexDirection": "row",
              "padding": 30,
            }
          }
        >
          <StyledButton
            onPress={[Function]}
            outlined={true}
            style={
              Object {
                "flex": 1,
                "marginRight": 30,
              }
            }
          >
            Disagree
          </StyledButton>
          <StyledButton
            disabled={true}
            onPress={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Agree
          </StyledButton>
        </Styled(Component)>,
      ],
      "style": Object {
        "marginTop": 20,
        "paddingHorizontal": 25,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Your consent",
          "size": "large",
          "style": Object {
            "marginBottom": 15,
          },
        },
        "ref": null,
        "rendered": "Your consent",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledText
              weight={true}
            >
              Don't worry, this is all perfectly safe.
            </StyledText>,
            <StyledText
              weight={true}
            >
              All we can do is look at...
            </StyledText>,
          ],
          "style": Object {
            "flex": 1,
            "marginBottom": 15,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Don't worry, this is all perfectly safe.",
              "weight": true,
            },
            "ref": null,
            "rendered": "Don't worry, this is all perfectly safe.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "All we can do is look at...",
              "weight": true,
            },
            "ref": null,
            "rendered": "All we can do is look at...",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <AccordionView />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <StyledText
            style={
              Object {
                "marginVertical": 15,
              }
            }
          >
            gameplan can only read your bank data and can't make payments or change your account.
          </StyledText>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "gameplan can only read your bank data and can't make payments or change your account.",
            "style": Object {
              "marginVertical": 15,
            },
          },
          "ref": null,
          "rendered": "gameplan can only read your bank data and can't make payments or change your account.",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <StyledText
            style={
              Object {
                "marginBottom": 15,
              }
            }
          >
            We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission
          </StyledText>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission",
            "style": Object {
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": "We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Component)
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Switch
                disabled={false}
                onValueChange={[Function]}
                value={false}
              />
            </Styled(Component)>,
            <StyledText
              style={
                Object {
                  "flex": 2,
                  "fontSize": 15,
                  "marginBottom": 15,
                }
              }
            >
              You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data
            </StyledText>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Switch
                disabled={false}
                onValueChange={[Function]}
                value={false}
              />,
              "style": Object {
                "alignItems": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "onValueChange": [Function],
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data",
              "style": Object {
                "flex": 2,
                "fontSize": 15,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": "You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <StyledText
            style={
              Object {
                "marginBottom": 15,
              }
            }
          >
            We will now securely take you to your bank/building society
          </StyledText>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "We will now securely take you to your bank/building society",
            "style": Object {
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": "We will now securely take you to your bank/building society",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledButton
              onPress={[Function]}
              outlined={true}
              style={
                Object {
                  "flex": 1,
                  "marginRight": 30,
                }
              }
            >
              Disagree
            </StyledButton>,
            <StyledButton
              disabled={true}
              onPress={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              Agree
            </StyledButton>,
          ],
          "style": Object {
            "flexDirection": "row",
            "padding": 30,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Disagree",
              "onPress": [Function],
              "outlined": true,
              "style": Object {
                "flex": 1,
                "marginRight": 30,
              },
            },
            "ref": null,
            "rendered": "Disagree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Agree",
              "disabled": true,
              "onPress": [Function],
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": "Agree",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StyledText
            size="large"
            style={
              Object {
                "marginBottom": 15,
              }
            }
          >
            Your consent
          </StyledText>,
          <Component
            style={
              Object {
                "flex": 1,
                "marginBottom": 15,
              }
            }
          >
            <StyledText
              weight={true}
            >
              Don't worry, this is all perfectly safe.
            </StyledText>
            <StyledText
              weight={true}
            >
              All we can do is look at...
            </StyledText>
          </Component>,
          <Component>
            <AccordionView />
          </Component>,
          <Component>
            <StyledText
              style={
                Object {
                  "marginVertical": 15,
                }
              }
            >
              gameplan can only read your bank data and can't make payments or change your account.
            </StyledText>
          </Component>,
          <Component>
            <StyledText
              style={
                Object {
                  "marginBottom": 15,
                }
              }
            >
              We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission
            </StyledText>
          </Component>,
          <Styled(Component)>
            <Styled(Component)
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Switch
                disabled={false}
                onValueChange={[Function]}
                value={false}
              />
            </Styled(Component)>
            <StyledText
              style={
                Object {
                  "flex": 2,
                  "fontSize": 15,
                  "marginBottom": 15,
                }
              }
            >
              You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data
            </StyledText>
          </Styled(Component)>,
          <Component>
            <StyledText
              style={
                Object {
                  "marginBottom": 15,
                }
              }
            >
              We will now securely take you to your bank/building society
            </StyledText>
          </Component>,
          <Styled(Component)
            style={
              Object {
                "flexDirection": "row",
                "padding": 30,
              }
            }
          >
            <StyledButton
              onPress={[Function]}
              outlined={true}
              style={
                Object {
                  "flex": 1,
                  "marginRight": 30,
                }
              }
            >
              Disagree
            </StyledButton>
            <StyledButton
              disabled={true}
              onPress={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              Agree
            </StyledButton>
          </Styled(Component)>,
        ],
        "style": Object {
          "marginTop": 20,
          "paddingHorizontal": 25,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Your consent",
            "size": "large",
            "style": Object {
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": "Your consent",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledText
                weight={true}
              >
                Don't worry, this is all perfectly safe.
              </StyledText>,
              <StyledText
                weight={true}
              >
                All we can do is look at...
              </StyledText>,
            ],
            "style": Object {
              "flex": 1,
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Don't worry, this is all perfectly safe.",
                "weight": true,
              },
              "ref": null,
              "rendered": "Don't worry, this is all perfectly safe.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "All we can do is look at...",
                "weight": true,
              },
              "ref": null,
              "rendered": "All we can do is look at...",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <AccordionView />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <StyledText
              style={
                Object {
                  "marginVertical": 15,
                }
              }
            >
              gameplan can only read your bank data and can't make payments or change your account.
            </StyledText>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "gameplan can only read your bank data and can't make payments or change your account.",
              "style": Object {
                "marginVertical": 15,
              },
            },
            "ref": null,
            "rendered": "gameplan can only read your bank data and can't make payments or change your account.",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <StyledText
              style={
                Object {
                  "marginBottom": 15,
                }
              }
            >
              We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission
            </StyledText>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission",
              "style": Object {
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": "We connect securely to your bank through our trusted partner TrueLayer and we'll only be able to view these for three months. Your data will never be shared with third parties without your permission",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Component)
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Switch
                  disabled={false}
                  onValueChange={[Function]}
                  value={false}
                />
              </Styled(Component)>,
              <StyledText
                style={
                  Object {
                    "flex": 2,
                    "fontSize": 15,
                    "marginBottom": 15,
                  }
                }
              >
                You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data
              </StyledText>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Switch
                  disabled={false}
                  onValueChange={[Function]}
                  value={false}
                />,
                "style": Object {
                  "alignItems": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "onValueChange": [Function],
                  "value": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data",
                "style": Object {
                  "flex": 2,
                  "fontSize": 15,
                  "marginBottom": 15,
                },
              },
              "ref": null,
              "rendered": "You agree to our terms & conditions and privacy policy which covers how giffgaff can access and use your data",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <StyledText
              style={
                Object {
                  "marginBottom": 15,
                }
              }
            >
              We will now securely take you to your bank/building society
            </StyledText>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "We will now securely take you to your bank/building society",
              "style": Object {
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": "We will now securely take you to your bank/building society",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledButton
                onPress={[Function]}
                outlined={true}
                style={
                  Object {
                    "flex": 1,
                    "marginRight": 30,
                  }
                }
              >
                Disagree
              </StyledButton>,
              <StyledButton
                disabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                Agree
              </StyledButton>,
            ],
            "style": Object {
              "flexDirection": "row",
              "padding": 30,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Disagree",
                "onPress": [Function],
                "outlined": true,
                "style": Object {
                  "flex": 1,
                  "marginRight": 30,
                },
              },
              "ref": null,
              "rendered": "Disagree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Agree",
                "disabled": true,
                "onPress": [Function],
                "style": Object {
                  "flex": 1,
                },
              },
              "ref": null,
              "rendered": "Agree",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
